### ISSUE TITLE
Critical security issue: Path traversal vulnerability in Core.php

### ISSUE THREAD

**Issue by predakanga** [davidjconnelly/trongate-framework#39]
In engine/Core.php, you read and display a file to the user agent without any form of authentication or validation: https://github.com/davidjconnelly/trongate-framework/blob/master/engine/Core.php#L13-L28

This file can be any file that the executing user (i.e. the apache2 or php-fpm user) has read access to. This includes sensitive configuration files such as config/db.php or your encryption class and accompanying key. Depending on the operating system's setup, this may also put the underlying system at risk of compromise.

The ideal behaviour here would be to remove the relevant branch entirely and simply place assets in the web root such that the web server can serve them without invoking PHP.

Failing that, you should pass all given URLs through realpath and check them against an allowlist.

**Comment by predakanga** [comment]
Additionally, please consider changing the relevant passwords, keys, etc on trongate.io - I have confirmed that this installation is vulnerable at the time of writing.

**Comment by davidjconnelly** [comment]
Thank you for bringing this up.  I think you might have spotted a legitimate bug!

I've removed the code that you've highlighted and I've pushed out an update.  I'm also going to have a word with the guy who wrote the code for handling vendor/autoload tomorrow.  I'd like to see what he has to say.  

In the meantime, I think your bug report looks legitimate and I'm very grateful for that.

Thank you!
